shop/src/main/java/ro/msg/learning/dto/OrderDto.java

```diff
@@ -0,0 +1,21 @@
+package ro.msg.learning.dto;
+
+import java.time.LocalDateTime;
+import java.util.Map;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import ro.msg.learning.entity.Address;
+import ro.msg.learning.entity.Customer;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class OrderDto {
+	private LocalDateTime createdAt;
+	private Integer customerId;
+	private Address address;
+	// since JSON only allows String keys for maps, I have passed the product Id as String
+	private Map<String, Integer> productsAndCorrespondingQuantities;
```

Comment:
you could create an OrderDetailDto(productId, quantity) and use here a List<OrderDetailDto>

Reply:
Fixed